### DISCUSSION TITLE
French translation

### DISCUSSION THREAD

**Discussion by NathanBnm** [breezy-weather/breezy-weather#317] [Q&A] [answered]
I would like to contribute to French translation but Weblate does not allow me to add the French language. Could you please add it for me?

**Comment by papjul** [comment]
Je suis le traducteur de la version française de Breezy Weather, que voudrais-tu changer dans la traduction ?

**Reply by NathanBnm** [reply]
@papjul je ne savais pas que la traduction française existait déjà justement elle n'apparaît pas dans Weblate

**Reply by papjul** [reply]
Comment ça ? L'application devrait s'afficher directement en français quand tu l'ouvres, si tu as bien défini la langue de ton appareil en Français, sinon tu peux aller dans Settings > Appearance > Languages et forcer le français.

Quand j'ajoute une nouvelle chaîne de caractères, je la traduis systématiquement en anglais et en français, donc rien ne manque, il n'est donc pas prévu de l'ouvrir à la traduction sur Weblate.

**Reply by NathanBnm** [reply]
Je n'utilise pas personnellement l'application je l'ai simplement vue sur Weblate et je souhaitais y contribuer.

Au contraire le fait de ne pas mettre la traduction en français sur Weblate donne la fausse impression qu'elle n'existe pas. Il vaut mieux ouvrir la traduction sur Weblate même si elle est complète. De cette façon les contributeurs pourront proposer des corrections et compléter les nouvelles chaînes qui seront ajoutées. Il est toujours possible de mettre les traductions validées en lecture seule.

**Reply by papjul** [reply]
Je sais que l’intention est bienveillante, mais je pense vraiment qu’une application ne devrait pas être traduite par quelqu’un qui ne l’utilise pas, tu passes à côté du contexte dans lequel les chaînes de caractère sont utilisées :/

Je suis d’accord que ce n’est pas propre comment c’est fait actuellement, mais c’est tout simplement car c’est le seul moyen que j’ai trouvé, et ça semble être le seul moyen préconisé dans la doc pour filtrer les langues ouvertes à la traduction ("It can be used to limit the list of languages managed by Weblate.") :
https://docs.weblate.org/en/latest/admin/projects.html#component-language-regex

Je veux bien que tu m’indiques comment mettre une langue en lecture seule, car je n’ai pas trouvé, et je ne trouve rien avec le mot-clé "read-only" dans la doc.

**Reply by NathanBnm** [reply]
Si j'avais la possibilité de participer à la traduction j'aurai bien sûr installé l'application pour avoir le contexte. Je suis déjà contributeur sur plusieurs projets.

En fait il est possible d'activer la révision des traductions et d'approuver les chaînes une à une (https://docs.weblate.org/en/latest/workflows.html#translation-states). Dès lors qu'une chaîne est approuvée les traducteurs ne peuvent plus la modifier et uniquement proposer des suggestions. Je trouve que ce serait la meilleure approche ici.

**Reply by papjul** [reply]
Si je comprends bien, cela m’oblige à approuver les 775 chaînes de caractère une par une, et à chaque fois que je rajoute une traduction, aller sur Weblate et aller approuver. Et cela n’empêche personne de rejoindre l’équipe “review” et réécrire derrière. Ce n’est pas une solution viable. Je vais poser la question à Weblate directement, je te tiendrai au courant.

**Reply by papjul** [reply, marked as answer]
Merci pour le signalement, j'ai demandé à Weblate quel était le meilleur moyen de gérer ça, et ils m'ont indiqué qu'il s'agissait en réalité d'un bug. Il vient d'être corrigé donc cela sera en place prochainement et la traduction française ne sera désormais plus proposée comme suggestion.
